### DISCUSSION TITLE
Support modifying vercel.json during build step

### DISCUSSION THREAD

**Discussion by vinaypuppal** [vercel/community#3323] [Idea]
I am trying to add some headers to the `vercel.json` file in my build step but they were not picked up by Vercel.  I am writing the vercel.json file to the root of the project, so do I need to place the modified vercel.json file in some other directory or this is not supported?

**Comment by mcsdevv** [comment]
Hi @vinaypuppal, this is not something that's possible currently but may be added in the future.

If using Next.js, you can do this with the `next.config.js` file for most properties.

**Reply by vinaypuppal** [reply]
Thanks, @mcsdevv for the reply.

No, I am not using Next.js

We are building a more [generalized tool](https://github.com/rapidsec-com/security-headers-generator-jamstack) to add security headers to static sites, it would be helpful if Vercel implements this feature soon.

**Reply by shaialon** [reply]
@mcsdevv - it would be really great if Vercel can match other technologies like Firebase and Netlify, to support this feature.

**Comment by orta** [comment]
For folks looking for this, I think (will try next week) you can use the build output API instead: https://vercel.com/docs/build-output-api/v3

**Comment by amyegan** [comment]
Hey folks! This discussion is a couple of years old now, so there have certainly been some changes since it was originally posted. 

[Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) available now. It lets you do things, like modifying headers or redirecting routes, in a more dynamic way than could be done with vercel.json configuration.
